tests/Models/DiscussionModelTest.php

```diff
@@ -113,4 +121,91 @@ public function provideIsArchivedTests(): array {
 
         return $r;
     }
+
+
+    /**
+     * Test canClose() where Admin is false and user has CloseOwn permission.
+     */
+    public function testCanCloseAdminFalseCloseOwnTrue() {
+        $this->session->UserID = 123;
+        $this->session->getPermissions()->set('Vanilla.Discussions.CloseOwn', $this->session->UserID);
+        $this->session->getPermissions()->setAdmin(false);
+        $discussion = [
+            'DiscussionID' => 0,
+            'CategoryID' => 1,
+            'Name' => 'test',
+            'Body' => 'discuss',
+            'InsertUserID' => 123
+        ];
+        $actual = $this->model->canClose($discussion);
+        $expected = true;
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
+     * Test canClose() where Admin is false and user has CloseOwn permission but user did not start the discussion.
+     */
+    public function testCanCloseCloseOwnTrueNotOwn() {
+        $this->session->UserID = 123;
+        $this->session->getPermissions()->set('Vanilla.Discussions.CloseOwn', $this->session->UserID);
+        $this->session->getPermissions()->setAdmin(false);
+        $discussion = [
+            'DiscussionID' => 0,
+            'CategoryID' => 1,
+            'Name' => 'test',
+            'Body' => 'discuss',
+            'InsertUserID' => 321
+        ];
+        $actual = $this->model->canClose($discussion);
+        $expected = false;
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
+     * Test canClose() with discussion already closed and user didn't start the discussion.
+     */
+    public function testCanCloseCloseIsClosed() {
+        $this->session->UserID = 123;
+        $this->session->getPermissions()->set('Vanilla.Discussions.CloseOwn', $this->session->UserID);
+        $this->session->getPermissions()->setAdmin(false);
+        $discussion = [
+            'DiscussionID' => 0,
+            'CategoryID' => 1,
+            'Name' => 'test',
+            'Body' => 'discuss',
+            'InsertUserID' => 321,
+            'Closed' => true,
+            'Attributes' => ['ClosedByUserID' => 321]
+        ];
+        $actual = $this->model->canClose($discussion);
+        $expected = false;
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
+     * Test canClose() where Admin is true.
+     */
+    public function testCanCloseAdminTrue() {
+        $this->session->UserID = 123;
+        $discussion = ['DiscussionID' => 0, 'CategoryID' => 1, 'Name' => 'test', 'Body' => 'discuss', 'InsertUserID' => 123];
+        $actual = $this->model->canClose($discussion);
+        $expected = true;
+        $this->assertSame($expected, $actual);
+    }
+
+    /**
+     * Test canClose() with discussion object.
+     */
+    public function testCanCloseDiscussionObject() {
+        $this->session->UserID = 123;
+        $discussion = new \stdClass();
```

Comment:
As an FYI: You can convert between a stdClass and array like this:

```php
$discussion = (object)['DiscussionID' => 0, 'CategoryID' => 1, 'Name' => 'test', 'Body' => 'discuss', 'InsertUserID' => 123];
```

This would allow you to cut down on code between your object/array tests.

Reply:
This isn't me requesting a change. Just some info,

Reply:
Good to know.